### PR TITLE
[Blog] Muon Optimizer Support in DeepSpeed

### PR DESCRIPTION
**Pull request by delock**
Author: @PKUWZP & @delock 
Blog post introducing Muon optimizer support in DeepSpeed, covering how it integrates with
ZeRO Stage 2/3, measured convergence and memory results, and the roadmap ahead.

### CODE REVIEW

**Comment by delock** [comment]
Hi @PKUWZP @tjruwase this PR is ready for review and merge, thanks!

File: blogs/muon-optimizer/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Using Muon Optimizer with DeepSpeed
+## TL;DR
+Muon optimizer has gained momentum with more and more use from community and also from Large Foundation Model like Kimi-K2-Thinking.  Now DeepSpeed supports Muon optimizer.
+
+## What is Muon optimizer?
+Muon is an optimizer designed for hidden 2D weights of a neural network.  It takes gradient of the weight, computes its momentum, and applies Newton-Schulz iterations to orthogonalize the momentum matrix, then uses this orthogonalized matrix to update the weight[1](https://kellerjordan.github.io/posts/muon/).  Because Muon only maintains one momentum buffer (versus Adam’s two), it uses less memory for optimizer states.
```
Comment (sfc-gh-truwase) [inline blogs/muon-optimizer/README.md:6]: The citations are appear broken when rendered. For example, see `weight[1]` below. 

Image

File: blogs/muon-optimizer/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- ZeRO Stage 2, bf16
+- Batch size: 32 (4 per GPU), 8 GPUs (A100 40GB)
+- 1 epoch (~1460 steps), eval every 100 steps
+- Muon lr: 5e-3, Adam lr: 5e-6
+- LR schedule: constant (no warmup, no decay)
+- Gradient clipping: 1.0
+
+![Muon optimizer convergence on Qwen2.5-3B](images/muon_loss_3b.png)
+
+Muon optimizer converges smoothly and shows no overfitting during finetuning.
```
Comment (sfc-gh-truwase) [inline blogs/muon-optimizer/README.md:43]: Can we show side-by-side convergence comparison with AdamW?

File: blogs/muon-optimizer/README.md (hunk 1 above)
Comment (sfc-gh-truwase) [inline blogs/muon-optimizer/README.md:6]: I think this phrase, `hidden 2D weights`, requires more introduction and description. I think this section should give the high-level memory efficiency advantage of Muon over Adam before diving into implementation details.

Comment (delock) [reply blogs/muon-optimizer/README.md]: Hi @sfc-gh-truwase, do you suggest the citateion to have brackets appear around the citation number after rendering, while the link follows click of the number?

File: blogs/muon-optimizer/README.md (hunk 2 above)
Comment (delock) [reply blogs/muon-optimizer/README.md]: Yes, we have latest muon vs. adamw number on a muon pretrained model, will update soon

**Comment by delock** [comment]
Hi @sfc-gh-truwase @PKUWZP The PR had been updated according to comments.  Specifically, for AdamW vs. Muon comparison, we changed to Moonlight 16B A3B, which is a MoE model pretrained with Muon optimizer.   We finetune it with CodeAlpaca and benchmark the checkpoint with MBPP/MBPP+.   Also address the reference rendering issue and other comments.  See the updates for detail.  Thanks!

File: blogs/muon-optimizer/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+- [x] **ZeRO Stage 2 support** — merged
+- [x] **ZeRO Stage 3 support** — merged
+- [x] **Gram-Schmidt based Newton-Schulz iteration** — a faster orthogonalization kernel, in review
+- [ ] **CPU Offloading** — in progress
+- [ ] **MuonClip** — the variant used by Kimi-K2, planned
+
+If you have thoughts, feedback, or contributions on Muon optimizer, welcome to start an issue for discussion or submit a PR to DeepSpeed.  Let’s make Muon rock solid and lightning fast in DeepSpeed!
+
+## Contributors
+This work is contributed from Wang, Zhipeng (@PKUWZP); Chi McIsaac(@qimcis) and Ma, Guokai (@delock)
```
Comment (PKUWZP) [inline blogs/muon-optimizer/README.md]: We should add @pengdurice to the authorship as well.

Comment (delock) [reply blogs/muon-optimizer/README.md]: @PKUWZP done.  Is the full name correct? @pengdurice

**Review by sfc-gh-truwase — APPROVED** [review]
